transform/util.go

```diff
@@ -133,6 +129,82 @@ func getJSONRaw(data []byte, path string, pathRequired bool) ([]byte, error) {
 	return result, nil
 }
 
+// setPath updates the value with properly formatted timestamp(s) and properly
+// handles array indexing
+func setJSONRaw(data, out []byte, path string) ([]byte, error) {
+	var err error
+	splitPath := strings.Split(path, ".")
+	numOfInserts := 0
+
+	for element, k := range splitPath {
+		arrayRefs := jsonPathRe.FindAllStringSubmatch(k, -1)
+		if arrayRefs != nil && len(arrayRefs) > 0 {
+			objKey := arrayRefs[0][1]      // the key
+			arrayKeyStr := arrayRefs[0][2] // the array index
+			if arrayKeyStr == "*" {
```

Comment:
Yeah good point, I'll add the comment. My goal was to make this a generalized function for setting arbitrary data, so I wanted to ensure everything was supported out of the gate.